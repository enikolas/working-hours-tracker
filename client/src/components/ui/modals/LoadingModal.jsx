import React from 'react';
import PropTypes from 'prop-types';

import Modal from '../Modal';
import Spinner from '../Spinner';
import strings from '../../../utils/strings';

import '../../ui/FullScreenSpinner.css';

const LoadingModal = props => (
  <Modal
    active={props.active}
    title={props.title ? props.title : strings.pageLoading}
    content={<Spinner class="loadingModal" />}
    hasButtons={false}
  />
);

LoadingModal.propTypes = {
  active: PropTypes.bool,
  title: PropTypes.string
};

LoadingModal.defaultProps = {
  active: false,
  title: null
};

export default LoadingModal;
