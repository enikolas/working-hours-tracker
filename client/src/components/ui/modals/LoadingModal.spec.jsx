import React from 'react';
import { shallow } from 'enzyme';

import LoadingModal from './LoadingModal';

describe('LoadingModal', () => {
	it('should render properly without props', () => {
		const wrapper = shallow(<LoadingModal />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should render with a new title name', () => {
		const wrapper = shallow(<LoadingModal title="this is a loading modal title" />);
		expect(wrapper).toMatchSnapshot();
	});
});
