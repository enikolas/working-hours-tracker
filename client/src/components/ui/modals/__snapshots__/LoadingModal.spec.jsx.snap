// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingModal should render properly without props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingModal
    active={false}
    title={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": false,
      "buttons": Array [
        Object {
          "action": [Function],
          "label": "OK",
        },
      ],
      "cancel": null,
      "content": <Spinner
        class="loadingModal"
      />,
      "hasButtons": false,
      "title": "Carregando...",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": false,
        "buttons": Array [
          Object {
            "action": [Function],
            "label": "OK",
          },
        ],
        "cancel": null,
        "content": <Spinner
          class="loadingModal"
        />,
        "hasButtons": false,
        "title": "Carregando...",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`LoadingModal should render with a new title name 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingModal
    active={false}
    title="this is a loading modal title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": false,
      "buttons": Array [
        Object {
          "action": [Function],
          "label": "OK",
        },
      ],
      "cancel": null,
      "content": <Spinner
        class="loadingModal"
      />,
      "hasButtons": false,
      "title": "this is a loading modal title",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": false,
        "buttons": Array [
          Object {
            "action": [Function],
            "label": "OK",
          },
        ],
        "cancel": null,
        "content": <Spinner
          class="loadingModal"
        />,
        "hasButtons": false,
        "title": "this is a loading modal title",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
